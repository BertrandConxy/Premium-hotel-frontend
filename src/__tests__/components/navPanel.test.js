import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { screen, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import NavPanel from '../../components/NavPanel';
import store from '../../redux/configureStore';

const NavProvider = () => (
  <Provider store={store}>
    <Router>
      <NavPanel />
    </Router>
  </Provider>
);

describe('render component', () => {
  it('renders the rooms list form', () => {
    render(<NavProvider />);
    expect((screen.getByText(/home/i))).toBeInTheDocument();
    expect((screen.getByText(/my reservations/i))).toBeInTheDocument();
    expect((screen.getByText(/create room/i))).toBeInTheDocument();
    expect((screen.getByText(/delete room/i))).toBeInTheDocument();
    expect((screen.getByText(/reserve room/i))).toBeInTheDocument();
  });
});

describe('Snapshot test', () => {
  it('looks the same as the snapshot', () => {
    const component = renderer.create(<NavProvider />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
