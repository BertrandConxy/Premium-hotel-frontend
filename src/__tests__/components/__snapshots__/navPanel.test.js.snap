// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test looks the same as the snapshot 1`] = `
<section>
  <nav>
    <ul>
      <li
        className="nav-link the-nav pt-0 pb-0 float-left text-black"
      >
        <a
          activeclassname="active"
          aria-current="page"
          className="active"
          exact="true"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav-link the-nav pt-0 pb-0 float-left text-black"
      >
        <a
          activeclassname="active"
          className=""
          exact="true"
          href="/reservations"
          onClick={[Function]}
        >
          My Reservations
        </a>
      </li>
      <li
        className="nav-link the-nav pt-0 pb-0 float-left text-black"
      >
        <a
          activeclassname="active"
          className=""
          exact="true"
          href="/create_room"
          onClick={[Function]}
        >
          Create Room
        </a>
      </li>
      <li
        className="nav-link the-nav pt-0 pb-0 float-left text-black"
      >
        <a
          activeclassname="active"
          className=""
          exact="true"
          href="/rooms"
          onClick={[Function]}
        >
          Delete Room
        </a>
      </li>
      <li
        className="nav-link the-nav pt-0 pb-0 float-left text-black"
      >
        <a
          activeclassname="active"
          className=""
          exact="true"
          href="/add_reservation"
          onClick={[Function]}
        >
          Reserve Room
        </a>
      </li>
    </ul>
  </nav>
</section>
`;
